google-startup-scripts/usr/share/google/first-boot

```diff
@@ -32,41 +31,28 @@ function get_instance_id() {
   ${PREFIX}/usr/share/google/get_metadata_value id 2>/dev/null
 }
 
-# Output the instance id.
-# Returns 1 if the instance id hasn't changed, 0 if it has.
-function update_instanceid_on_disk() {
+# Checks the instance id has changed.
+# Exits with return code 0 if the instance id hasn't changed.
+function check_instanceid_on_disk() {
   local readonly instanceid=$1
 
   if [[ "${instanceid}" == "" ]]; then
-    # Cannot determine instance id.  Either we're not running on a Compute VM,
+    # Cannot determine instance id. Either we're not running on a Compute VM,
     # or networking hasn't started up yet, etc.
-    exit 1
+    exit 0
   fi
 
   if [[ "${instanceid}" != "unknown-instance" &&
         "${instanceid}" == "$(cat ${INSTANCE_FILE} 2>/dev/null)" ]]; then
     # Instance id is same as on disk.
-    exit 1
+    exit 0
   fi
-
-  # We are booting this instance for the first time.
-  echo ${instanceid} > ${INSTANCE_FILE}
-  exit 0
 }
 
-# This uses a subshell, opens LOCK_FILE on FD 200, locks it, and then
-# checks whether the instance has changed.
-function sync_new_instance_check() {
-  local instanceid=$(get_instance_id)
-  (flock 200; update_instanceid_on_disk ${instanceid}) 200> ${LOCK_FILE}
-  return $?
-}
+declare -r INSTANCE_ID=$(get_instance_id)
 
-sync_new_instance_check
-if [[ $? != 0 ]]; then
-  # The instance hasn't changed.  Just exit.
-  exit 0
-fi
+# Checks whether the instance has changed.
```

Comment:
Might help to add a note here that the function exits if the instance has already been booted.